prototype_source/maskedtensor_adagrad.rst

```diff
@@ -0,0 +1,182 @@
+Efficiency of writing "sparse" semantics for Adagrad
+====================================================
+
+`Issue 1369 <https://github.com/pytorch/pytorch/issues/1369>`__ discussed the additional lines of code
+that were introduce while writing "sparse" semantics for Adagrad.
+But really the code doesn't use sparsity as a compression and optimization technique,
+it wants to use masked semantics. We worked around this by introducing one-off semantics and operators
+that encode this behavior while forcing users to be aware of storage details such as indices and values.
+
+In particular we'll point out when sparsity is used as a semantic extension, i.e. unspecified values are not zero
+and when it is just used to compress zeros.
+We'll also compare and contrast this with equivalent code written using MaskedTensor.
+In the end the code snippets are repeat without additional comments to show the difference in brevity.
+
+Set up
+------
+
+.. code-block:: python
+
+    import torch
+    from torch.masked.maskedtensor import masked_tensor
```

Comment:
When I try to run this in colab, it cannot find torch.masked --> this is because torch.masked has not been released yet right?

Reply:
This would be cool to actually bring in when you use it. For the setup chunk and the adagrad current impl chunk, maskedtensor is not actually necessary yet. It could be a good idea to be like "now import maskedtensor and see the magical reduction and simplicity!"